src/mui/input/ReferenceArrayInput.js

```diff
@@ -172,33 +230,44 @@ export class ReferenceArrayInput extends Component {
             );
         }
 
-        if (!(referenceRecords && referenceRecords.length > 0) && !allowEmpty) {
+        const translatedLabel = translate(
+            label || `resources.${resource}.fields.${source}`
+        );
+
+        const dataStatus = getDataStatus({
+            input,
+            matchingReferences,
+            referenceRecords,
+            translate,
+        });
+
+        if (dataStatus.waiting) {
+            return <ReferenceLoadingProgress label={translatedLabel} />;
+        }
+
+        if (dataStatus.error) {
             return (
-                <Labeled
-                    label={
-                        typeof label === 'undefined' ? (
-                            `resources.${resource}.fields.${source}`
-                        ) : (
-                            label
-                        )
-                    }
-                    source={source}
-                    resource={resource}
+                <ReferenceError
+                    label={translatedLabel}
+                    error={translate('aor.input.references.all_missing', {
+                        _: 'aor.input.references.all_missing',
+                    })}
                 />
             );
         }
 
         return React.cloneElement(children, {
             allowEmpty,
             input,
-            label:
-                typeof label === 'undefined'
-                    ? `resources.${resource}.fields.${source}`
-                    : label,
+            label: translatedLabel,
             resource,
-            meta,
+            meta: {
+                ...meta,
+                error: dataStatus.warning,
```

Comment:
I'm afraid this may override actual errors from the filed level validator. Did you test it?

Reply:
After review, it's working.

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/547706/36200128-f1da53c8-117b-11e8-910f-a1d1f0226ed2.gif)

![ezgif com-video-to-gif 1](https://user-images.githubusercontent.com/547706/36200171-18b85864-117c-11e8-80c7-ef580cfab753.gif)